Homework done!

tweexmusic opened the pull request:
-Player.cs added
-Desired changes have been made
-See comments at bottom of Test.cs for specifics

Commit:
Homework done!

-Player.cs added
-Desired changes have been made
-See comments at bottom of Test.cs for specifics

Commit:
Creating EnemyManager and controlling inside Player.cs

Commit:
Created EnemyManager singleton

Commit:
Started GameManager code

Commit:
Coded the GameManager, Player singleton, and enemy attacks

-GameManager prefab spawns instances of EnemyManager and Player prefabs
-Player has singleton for enemy attacks
-Created enemy attacks that deal damage to player.

Commit:
Code cleanup

Commit:
FMOD integration, new CS files to control it

-FMOD integration (plugin version 2.00.10)
-New prefabs for FMOD control
-Added background music and sound effects for each player/enemy attack.

tweexmusic committed:
Music Level 1 Enum Test

Created enum for battle parameter.

tweexmusic committed:
Changed FMOD settings to store .banks inside project

nshooter3 commented on Assets/Scripts/GameManager.cs:
```diff
@@ -0,0 +1,53 @@
+    {
+        if (Player.instance == null)
+        {
+            Player.instance = Instantiate(playerPrefab);
```
You actually don't need to set the class instances here, since each of your classes does that itself in their awake functions. This isn't wrong, but it is redundant. This applies to all three of your instantiation functions.

tweexmusic replied:
Did not remove these from the code base.  When I did, the null object error appeared.  I'm guessing there is something I'm missing.

nshooter3 commented on Assets/Scripts/EnemyManager.cs:
```diff
@@ -0,0 +1,72 @@
+        }
+
+
+        void PopulateEnemyList()
```
What's going on here, why did you put a function inside another function? I'm surprised that's allowed, lol. Anyway, move this outside of your awake function.

nshooter3 commented on Assets/Scripts/EnemyManager.cs:
```diff
@@ -0,0 +1,72 @@
+    public Enemy koopaPrefab;
+    public Enemy bobombPrefab;
+
+    public string enemyName;
```
You can delete this variable, it isn't actually doing anything. The enemy name param comes from the enemy base class, not the enemy manager.

nshooter3 commented on Assets/Scripts/Player.cs:
```diff
@@ -0,0 +1,58 @@
+    {
+        FMODOneShotPlayer.instance.FMODPlayOneShotSound("event:/sfx/abilities/hamster_shoot");
+
+        foreach (Enemy enemy in EnemyManager.instance.enemiesList)
```
This part of the function where you iterate through enemiesList would make more sense as a separate function in the enemy manager. Making enemiesList public and letting the player iterate through it is a bit messy, generally you want to avoid classes digging through other classes' data structures when you can. The enemy manager should be the interface through which the other scripts interact with the enemy list, and enemyList itself should be private.

tweexmusic committed:
Revisions made

-Revisions based on comments have been made.
-Refactoring of Enemy and Player attack/takedamage functions
-Other small cleanup

tweexmusic committed:
Removed Class.instance from instantiation statement
